src/include/win32/fs_compat.h

```diff
@@ -23,6 +23,10 @@
 #define S_ISGID     02000
 #define S_ISVTX     01000
 
+#ifndef ALLPERMS
+#define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)
```

Comment:
This is not specific to win32, so placing this in a win32-specific header is probably wrong.  Looking at https://github.com/ceph/ceph/commit/6ee073234cc1361309716713bbbb773005bda85d, the reason this define was added to `src/test/libcephfs/test.cc` is that some libc implementations on Linux miss it too.

I think a better fix would be to define `ALLPERMS` in some common header or just move the now unused definition in `src/test/libcephfs/test.cc` to `src/test/libcephfs/suidsgid.cc`.

Reply:
Makes sense. ``compat.h`` already defines ``ACCESSPERMS`` [1], we can move it there so that we won't have to redefine it in every file that uses it.

https://github.com/ceph/ceph/blob/f66bdb87f73855966b42747c6519addadc7c7411/src/include/compat.h#L32

Reply:
Make sure to drop the redundant definition from `src/test/libcephfs/test.cc` while at it.